engine/src/main/java/com/redhat/cloud/notifications/db/repositories/SubscriptionRepository.java

```diff
@@ -17,23 +19,36 @@ public class SubscriptionRepository {
     @Inject
     EntityManager entityManager;
 
-    public List<String> getSubscribers(String orgId, UUID eventTypeId, SubscriptionType subscriptionType) {
-        return getSubscriptions(orgId, eventTypeId, subscriptionType, true);
+    public List<String> getSubscribers(String orgId, UUID eventTypeId, SubscriptionType subscriptionType, Optional<Severity> severity) {
+        return getSubscriptions(orgId, eventTypeId, subscriptionType, true, severity);
     }
 
-    public List<String> getUnsubscribers(String orgId, UUID eventTypeId, SubscriptionType subscriptionType) {
-        return getSubscriptions(orgId, eventTypeId, subscriptionType, false);
+    public List<String> getUnsubscribers(String orgId, UUID eventTypeId, SubscriptionType subscriptionType, Optional<Severity> severity) {
+        return getSubscriptions(orgId, eventTypeId, subscriptionType, false, severity);
     }
 
-    private List<String> getSubscriptions(String orgId, UUID eventTypeId, SubscriptionType subscriptionType, boolean subscribed) {
-        String hql = "SELECT id.userId FROM EventTypeEmailSubscription WHERE id.orgId = :orgId AND id.subscriptionType = :subscriptionType " +
+    private List<String> getSubscriptions(String orgId, UUID eventTypeId, SubscriptionType subscriptionType, boolean subscribed, Optional<Severity> severity) {
+        if (severity.isPresent()) {
+            String sql = "SELECT user_id FROM email_subscriptions WHERE org_id = :orgId AND subscription_type = :subscriptionType AND event_type_id = :eventTypeId "
+                + "AND ((severities is null AND subscribed = :subscribed) OR ((severities ->> :severity) = CAST(:subscribed AS TEXT)))";
```

Comment:
```suggestion
                + "AND ((severities is null AND subscribed IS :subscribed) OR ((severities ->> :severity) = CAST(:subscribed AS TEXT)))";
```

Reply:
Unrelated: Is the `CAST` really needed? We're serializing a `Boolean` value in `SubscriptionRepository`.